gst/tensor_decoder/tensordec.c

```diff
@@ -670,10 +681,22 @@ gst_tensordec_set_caps (GstBaseTransform * trans,
   silent_debug_caps (incaps, "from incaps");
   silent_debug_caps (outcaps, "from outcaps");
 
-  /** @todo Check if outcaps == getOutputDim (incaps) */
+  if (gst_tensordec_configure (self, incaps)) {
+    GstCaps *supposed = gst_tensordec_media_caps_from_tensor (self,
+        &self->tensor_config);
 
-  return TRUE;
+    /** Check if outcaps ==equivalent== supposed */
+    if (!gst_caps_is_always_compatible (outcaps, supposed)) {
+      GST_ERROR ("This is not compatible with the supposed output pad cap");
+      gst_caps_unref (supposed);
+      return FALSE;
+    }
+    gst_caps_unref (supposed);
+  } else {
+    return TRUE;
```

Comment:
Please change to FALSE if failed to get tensor info.

Reply:
Thanks! Fixed.